src/CppUTestExt/MockSupport_c.cpp

```diff
@@ -32,6 +32,8 @@
 #include "CppUTestExt/MockSupport.h"
 #include "CppUTestExt/MockSupport_c.h"
 
+typedef void Cpp_Func();  /* Cl2000 requires casts to C++ function */
+
```

Comment:
You objected to this typedef as well. It would have to be part of the package, but we might want it in a place where it is more generally accessible. BTW I tried all kinds of ways to cast the function pointer without a typedef and all compilers went beserk. So, if you can cast somehow other than with a typedef, that would be okay. But a typedef is cleaner in any case.

I know it sounds silly, but the cl2000 is fussy about this. Perhaps the warning can be disabled, if so, I'd go for that.